typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -250,11 +251,20 @@ async function sendMessage(
     logs: receipt.logs,
   });
 
-  await utils.timeout(
-    app.waitForMessageReceipt(receipt),
-    MESSAGE_RECEIPT_TIMEOUT,
-    'Timeout waiting for message to be received',
-  );
+  try {
+    await utils.timeout(
+      app.waitForMessageReceipt(receipt),
+      MESSAGE_RECEIPT_TIMEOUT,
+      'Timeout waiting for message to be received',
+    );
+  } catch (error) {
+    // If we weren't able to get the receipt for message processing, try to read the state to ensure it wasn't a transient provider issue
+    const statsAfter = await app.channelStats(origin, destination);
+    if (statsAfter.received === statsBefore.received) {
```

Comment:
```suggestion
    if (channelStatsNow.received <= channelStatsBefore.received) {
      log("Did not receive event for message delivery even though it was delivered", {origin, destination})
```

Reply:
I realize `<` should be impossible but it also makes it more clear when reading that it is a numeric comparison.